File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.py
import sys

n = int(sys.stdin.readline())
member_list = []

for i in range(n) :
    age, name = (sys.stdin.readline().rstrip().split())     #단어 입력
    member_list.append([int(age), name, int(i)])                      #리스트에 나이, 이름, 인덱스 저장

member_list.sort(key=lambda x:(x[0], x[2]))                 #나이[0] -> 인덱스[2] 정렬

for i in range(n) :
    print(member_list[i][0], member_list[i][1])             #나이[i][0], 이름[i][1] 출력

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.js
let fs = require("fs");
//let input = fs.readFileSync("./test.txt").toString().split("\n");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let n = Number(input[0]);
let arr = [];

for (let i = 1; i <= n; i++) {
    //onsole.log(input[i]);
    const age = Number(input[i].split(" ")[0]);
    const name = input[i].split(" ")[1];

    arr.push([age, name]);
}

const sortArr = arr.sort((a, b) => a[0] - b[0]);

let answer = "";
sortArr.forEach((item) => {
    answer += item[0] + " " + item[1] + "\n";
});

console.log(answer);
